File: algorithms/058.length-of-last-word.cpp
#include <boost/algorithm/stirng.hpp>

class Solution {
	public:
		void split(const std::string &s, char delim, std::vector<std::string> &elems) {
			std::stringstream ss; 
			ss.str(s);
			std::string item;
			while (std::getline(ss, item, delim)) {
				elems.push_back(item);
			}   
		}


		std::vector<std::string> split(const std::string &s, char delim) {
			std::vector<std::string> elems;
			split(s, delim, elems);
			return elems;
		}

		string trim(string s, char c){ 
			int start=0, end=s.size()-1;
			for (int i=0; i < s.size(); i++){
				if (s[i] == c)
					start++;
				else
					break;
			}   
			for (int i=end; i >= start && i >= 0; i--){
				if (s[i] == c)
					end--;
				else
					break;
			}   
			return s.substr(start, end-start+1);
		}

		int lengthOfLastWord(string s) {
			string trim_s = trim(s, ' '); 
			std::vector<std::string> x = split(trim_s, ' '); 
			if (x.size() > 0)
				return x[x.size()-1].size();
			else
				return 0;
		}

};

File: algorithms/058.length-of-last-word.go
func trim(s string, c rune) string {
    start, end := 0, len(s) - 1
    
    // Trim leading characters
    for start < len(s) && rune(s[start]) == c {
        start++
    }

    // Trim trailing characters
    for end >= start  && rune(s[end]) == c {
        end--
    }

    return s[start : end+1]
}

func lengthOfLastWord(s string) int {
    trimmed := trim(s, ' ')
    words := strings.Split(trimmed ," ")
    if len(words) > 0 {
        return len(words[len(words) - 1])    
    }
    return 0
}
